spec/models/incoming_mail_handler_spec.rb

```diff
@@ -1,9 +1,8 @@
 describe "Handles incoming email" do
```

Comment:
ok that makes some sense.

I'm fine with it being part of this PR as long as we have some rationale around why that code changed. Could you maybe annotate the commit just a little so that future us is clear about why all the tests got removed? It seems like a lot more deletion than just replacing `actions_for_approver` with `StepMailer.proposal_notification`